src/utils.ts

```diff
@@ -1,89 +1,124 @@
 import Web3 from 'web3';
 import { AVAILABLE_TLDS } from './constants';
+import { Utils } from './types';
+import { hash as namehash } from 'eth-ens-namehash';
+import { keccak256 } from 'js-sha3';
 
-/**
- * Checks if the contract in the given address has the given method 
- * 
- * @param web3 - Web3 instance 
- * @param contractAddress - address of the contract to check 
- * @param signatureHash - keccak256 of the method signature
- * 
- * @returns
- * true if method exists, false if not
- */
-export const hasMethod = async(web3: Web3, contractAddress: string, signatureHash: string) => {
-  const code = await web3.eth.getCode(contractAddress);
-  return code.indexOf(signatureHash.slice(2, signatureHash.length)) > 0;
-}
-
-/**
- * Checks if the given web3 instance has accounts to sign txs
- * 
- * @param web3 - Web3 instance 
- * 
- * @returns
- * true if has accounts, false if not
- */
-export const hasAccounts = async(web3: Web3) => {
-  let accounts = [];
-  try {
-    accounts = await web3.eth.getAccounts();
-  } catch {
-    return false;
+export default class implements Utils {
```

Comment:
What do you think of exporting utils just as pure functions and not a class. Then, `RNS` would build an object exporting Utils using each implementation